pkg/zson/raw.go

```diff
@@ -203,19 +249,32 @@ func zsonParseField(b []byte) ([]byte, []byte, error) {
 		}
 		return zval.AppendContainer(nil, vals), rest, nil
 	}
-	i := 0
+	to := 0
+	from := 0
 	for {
-		if i >= len(b) {
+		if from >= len(b) {
 			return nil, nil, ErrUnterminated
 		}
-		switch b[i] {
+		switch b[from] {
 		case semicolon:
-			return b[:i], b[i+1:], nil
+			if to == 1 && b[0] == '-' {
+				return nil, b[2:], nil
+			}
+			return b[:to], b[from+1:], nil
 		case backslash:
 			// XXX need to implement full escape parsing,
 			// for now just skip one character
-			i += 1
+			from += 1
+			e, n := unescape(b[from:])
```

Comment:
I think we should revise the ZSON spec so that `\\` and `\xdd` are the only escape sequences and then use pkg/zeek.Unescape here.

Reply:
+1

Reply:
zeek.Unescape allocates new memory whereas this code unescapes in place